### PR TITLE
Support lto on mingw

### PR DESCRIPTION
**Pull request by yujincheng08**

### CODE REVIEW

**Comment by rui314** [comment]
Some buildbots are broken. Please fix.

File: common/signal-win32.cc [hunk 1]
```diff
@@ -38,6 +38,10 @@ static LONG WINAPI vectored_handler(_EXCEPTION_POINTERS *exception_info) {
     static const char msg[] =
       "mold: failed to write to an output file. Disk full?\n";
     (void)!_write(_fileno(stderr), msg, sizeof(msg) - 1);
+  } else if (rec->ExceptionCode == EXCEPTION_STACK_OVERFLOW) {
+    static const char msg[] =
+      "mold: stackoverflow.\n";
+    (void)!_write(_fileno(stderr), msg, sizeof(msg) - 1);
```
Comment (rui314) [inline common/signal-win32.cc:44]: Unrelated change?

Comment (yujincheng08) [reply common/signal-win32.cc:44]: Yes. I encountered this problem during my testing and it makes me confused that `mold` exits with 1 without any message.

**Comment by yujincheng08** [comment]
@rui314 CI fixed

File: CMakeLists.txt [hunk 2]
```diff
@@ -56,6 +56,10 @@ if(MOLD_USE_MOLD)
   add_link_options(-fuse-ld=mold -Wl,--gdb-index)
 endif()
 
+if(MINGW)
+  set(CMAKE_DL_LIBS -ldl)
```
Comment (rui314) [inline CMakeLists.txt:60]: If this is equivalent to `target_link_libraries(mold PRIVATE dl)`, please do so instead of overriding the system-provided variable.

Comment (yujincheng08) [reply CMakeLists.txt]: Done
